src/wp-admin/setup-config.php

```diff
@@ -426,33 +426,56 @@ function setup_config_display_header( $body_classes = array() ) {
 })();
 </script>
 			<?php
-	else :
-		/*
-		 * If this file doesn't exist, then we are using the wp-config-sample.php
-		 * file one level up, which is for the develop repo.
-		 */
-		if ( file_exists( ABSPATH . 'wp-config-sample.php' ) ) {
-			$path_to_wp_config = ABSPATH . 'wp-config.php';
-		} else {
-			$path_to_wp_config = dirname( ABSPATH ) . '/wp-config.php';
-		}
+		else :
+			/*
+			 * If this file doesn't exist, then we are using the wp-config-sample.php
+			 * file one level up, which is for the develop repo.
+			 */
+			if ( file_exists( ABSPATH . 'wp-config-sample.php' ) ) {
+				$path_to_wp_config = ABSPATH . 'wp-config.php';
+			} else {
+				$path_to_wp_config = dirname( ABSPATH ) . '/wp-config.php';
+			}
 
-		$handle = fopen( $path_to_wp_config, 'w' );
-		foreach ( $config_file as $line ) {
-			fwrite( $handle, $line );
-		}
-		fclose( $handle );
-		chmod( $path_to_wp_config, 0666 );
-		setup_config_display_header();
-		?>
+			$error_message = '';
+			$handle        = fopen( $path_to_wp_config, 'w' );
+			if ( false !== $handle ) {
+				foreach ( $config_file as $line ) {
+					fwrite( $handle, $line );
+				}
+				fclose( $handle );
+			} else {
+				$wp_config_perms = fileperms( $path_to_wp_config );
+				if ( ! empty( $wp_config_perms ) && ! is_writable( $path_to_wp_config ) ) {
+					$wp_config_perms = substr( decoct( $wp_config_perms ), 2 );
+					$error_message   = sprintf(
+						/* translators: 1: wp-config.php, 2: wp-config.php file permission */
+						__( 'Unable to write to %1$s file due to the file permissions being too restrictive: %2$s. Please update the chmod file permissions to allow for write access.' ),
+						'<code>wp-config.php</code>',
+						"<code>{$wp_config_perms}</code>"
```

Comment:
I'm in two minds about including the current permission, happy to say the file is not writable as the visitor can install at this point. The thought behind the question is in case the site's owners messed up and made the file neither readable or writable by the web server user, eg `0600`, in which case it reveals something about the experience level of the person configuring the server.

In most cases, WP links to docs so that might help too:

https://github.com/WordPress/wordpress-develop/blob/2d7b7dd833070386525ac04af45d5e64d10b8937/src/wp-admin/includes/file.php#L321-L327

Maybe:

```php
printf(
	/* translators: 1 wp-config.php, 2: Documentation URL. */
	__( 'You need to make the file %1$s writable before you can save your changes. See <a href="%2$s">Changing File Permissions</a> for more information.' ),
	__( 'https://wordpress.org/support/article/changing-file-permissions/' )
);
```

As I say, I am in two minds so consider this a nit.

Reply:
I agree with you @peterwilsoncc. That string reads better and provides better guidance for users. Updated in commit https://github.com/WordPress/wordpress-develop/pull/722/commits/89d6689fd708fbe1e779a3b6dfcd6e1057ac6d8f